var/spack/repos/builtin/packages/xgboost/package.py

```diff
@@ -59,4 +62,24 @@ def cmake_args(self):
         if '@1.5: ^cuda@11.4:' in self.spec:
             args.append(self.define('BUILD_WITH_CUDA_CUB', True))
 
+        if self.spec.satisfies('+openmp%clang'):
+            OpenMP_C_FLAGS = "-fopenmp=libomp"
+            OpenMP_C_LIB_NAMES = "libomp"
+            args += [
+                self.define('OpenMP_C_FLAGS', OpenMP_C_FLAGS),
+                self.define('OpenMP_C_LIB_NAMES', OpenMP_C_LIB_NAMES),
+                self.define('OpenMP_CXX_FLAGS', OpenMP_C_FLAGS),
+                self.define('OpenMP_CXX_LIB_NAMES', OpenMP_C_LIB_NAMES),
+            ]
+            shared_lib_suffix = 'so'
+            if self.spec.satisfies('platform=darwin'):
+                shared_lib_suffix = 'dylib'
+            clang = self.compiler.cc
+            clang_bin = os.path.dirname(clang)
+            clang_root = os.path.dirname(clang_bin)
+            args += [
+                self.define('OpenMP_libomp_LIBRARY', clang_root +
+                            '/lib/libomp.' + shared_lib_suffix)
```

Comment:
Might be more reliable to use `find_library` here, then you just give it a root directory and a filename (without a suffix) and it will find the right file for you on any platform, even if it's in `lib64` instead of `lib`